File: 240609/그 계절, 그 날/that-season-that-day.py
def y_check(y):
    if y%4 == 0:
        if y%100 == 0:
            if y%400 == 0:
                return True
            else:
                return False
        else:
            return True
    else:
        return False

def f(y_, m, d):
    m_ = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    if y_ == True:
        m_[1] = m_[1] + 1

    if 1 <= m <= 12 and (m_[m-1] >= d):
        return True
    return False

def s(m):
    if 3 <= m <= 5:
        return 'Spring'
    elif 6 <= m <= 8:
        return 'Summer'
    elif 9 <= m <= 11:
        return 'Fall'
    else:
        return 'Winter'

y, m, d = tuple(map(int, input().split()))
c = False
if y_check(y):
    c = f(True, m, d)
else:
    c = f(False, m, d)

if c:
    print(s(m))
else:
    print(-1)

File: 250102/그 계절, 그 날/that-season-that-day.cpp
#include <iostream>
using namespace std;

int LastDayNumber(int y, int m) {
    if(m == 2) {
        if (y%4 == 0) {
            if (y%100 == 0 && y%400 == 0) {
                return 29;
            }
            if (y%100 == 0) {
                return 28;
            }
            return 29;
        } else {
            return 28;
        }
    }
    if(m == 4 || m == 6 || m == 9 || m == 11)
        return 30;
    
    return 31;
}

bool JudgeDay(int y, int m, int d) {
    if(m <= 12 && d <= LastDayNumber(y, m))
        return true;
    
    return false;
}

int main() {
    // 여기에 코드를 작성해주세요.
    int Y, M, D;
    cin >> Y >> M >> D;

    if (JudgeDay(Y, M, D)) {
        if (M <= 2 || M == 12) {
            cout << "Winter";
        } else if (M <= 5) {
            cout << "Spring";
        } else if (M <= 8) {
            cout << "Summer";
        } else {
            cout << "Fall";
        }
    } else {
        cout << -1;
    }
    return 0;
}